src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        int currentSize = size;
+        if (currentSize == array.length) {
+            array = (T[]) resize();
+        }
```

Comment:
You can move this conditional to method resize() and avoid code duplicates

Reply:
I created a new method just to check the size first, is it ok?